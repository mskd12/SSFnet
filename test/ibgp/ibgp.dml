# 
# ibgp.dml
#
# Three ASes connected in a line, the middle one containing two
# BGP routers, the others just one each.
#

_schema [ _find .schemas.Net ]

Net [  # the all-encompassing Net

  frequency 1000000000   # nanosecond simulation frequency

  bgpoptions [
    validation_test 8     # prints messages associated with this test
  ]

  # ---------------------------------------------------------------- Net 1
  Net [ # 1 router
    id 1
    AS_status boundary
    router [
      id 1
      graph [
        ProtocolSession [
          name bgp  use  SSF.OS.BGP4.BGPSession
          autoconfig false
          connretry_time 120  min_as_orig_time 15
          reflector false
          neighbor [ as 2  address 1(2)  use_return_address 1(1)
                     _extends .basic_ebgp_neighbor ]
        ]
        ProtocolSession [ name socket use SSF.OS.Socket.socketMaster ]
        ProtocolSession [ name tcp    use SSF.OS.TCP.tcpSessionMaster ]
        ProtocolSession [ name ip     use SSF.OS.IP ]
      ]
      interface [ id 0  virtual true ]
      interface [ id 1 ]
    ]
  ]

  # ---------------------------------------------------------------- Net 2
  Net [ # 2 routers
    id 2
    AS_status boundary
    router [
      id 1
      graph [
        ProtocolSession [
          name bgp  use  SSF.OS.BGP4.BGPSession
          autoconfig false
          connretry_time 120  min_as_orig_time 15
          reflector false
          # The reason we do not use virtual interface addresses for our
          # internal neighbor is because we are not running any internal
          # routing protocol (such as OSPF).  Normally, we would be, but
          # for the purposes of simplifying this validation test, it is
          # has been omitted.
          neighbor [ as 2  address 2(1)  use_return_address 1(1)
                     _extends .basic_ibgp_neighbor ] # internal
          neighbor [ as 1  address 1(1)  use_return_address 1(2)
                     _extends .basic_ebgp_neighbor ]
        ]
        ProtocolSession [ name socket use SSF.OS.Socket.socketMaster ]
        ProtocolSession [ name tcp    use SSF.OS.TCP.tcpSessionMaster ]
        ProtocolSession [ name ip     use SSF.OS.IP ]
      ]
      interface [ id 0  virtual true ]
      interface [ idrange [ from 1 to 2 ] ]
    ]
    router [
      id 2
      graph [
        ProtocolSession [
          name bgp  use  SSF.OS.BGP4.BGPSession
          autoconfig false
          connretry_time 120  min_as_orig_time 15
          reflector false
          # The reason we do not use virtual interface addresses for our
          # internal neighbor is because we are not running any internal
          # routing protocol (such as OSPF).  Normally, we would be, but
          # for the purposes of simplifying this validation test, it is
          # has been omitted.
          neighbor [ as 2  address 1(1)  use_return_address 2(1)
                     _extends .basic_ibgp_neighbor ] # internal
          neighbor [ as 3  address 1(1)  use_return_address 2(2)
                     _extends .basic_ebgp_neighbor ]
        ]
        ProtocolSession [ name socket use SSF.OS.Socket.socketMaster ]
        ProtocolSession [ name tcp    use SSF.OS.TCP.tcpSessionMaster ]
        ProtocolSession [ name ip     use SSF.OS.IP ]
      ]
      interface [ id 0  virtual true ]
      interface [ idrange [ from 1 to 2 ] ]
    ]
    link [ attach 1(1) attach 2(1) delay 0.001 ]  # router to router
  ]

  # ---------------------------------------------------------------- Net 3
  Net [ # 1 router
    id 3
    AS_status boundary
    router [
      id 1
      graph [
        ProtocolSession [
          name bgp  use  SSF.OS.BGP4.BGPSession
          autoconfig false
          connretry_time 120  min_as_orig_time 15
          reflector false
          neighbor [ as 2  address 2(2)  use_return_address 1(1)
                     _extends .basic_ebgp_neighbor ]
        ]
        ProtocolSession [ name socket use SSF.OS.Socket.socketMaster ]
        ProtocolSession [ name tcp    use SSF.OS.TCP.tcpSessionMaster ]
        ProtocolSession [ name ip     use SSF.OS.IP ]
      ]
      interface [ id 0  virtual true ]
      interface [ id 1 ]
    ]
  ]

  # ----------------------------------------------------------------
  # inter-AS links
  link [ attach 1:1(1) attach 2:1(2) delay 0.01 ]
  link [ attach 2:2(2) attach 3:1(1) delay 0.01 ]

]
