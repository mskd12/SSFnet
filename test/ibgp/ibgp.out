~#
~# = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = =
~# IBGP Validation Test: success if 1 internal and 2 external msgs rcvd
~# = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = =
~#
~# AS Z: received external update message from AS Y
~# AS Y, router 2: received internal update
~# AS Z: received external update message from AS Y
~#             *** TEST SUCCESSFUL ***
~#
